src/app/provider/view.nim

```diff
@@ -17,8 +22,85 @@ QtObject:
     result.status = status
     result.setup
 
-  proc postMessage*(self: Web3ProviderView, data: string): string {.slot.} =
-    # TODO: implement code from status-react/src/status_im/browser/core.cljs
-    echo "==========================================="
-    echo "Message received from JS web3 provider: ", data
-    return "Hello World!" # This can only be seen in chrome devtools
\ No newline at end of file
+  proc web3AsyncReadOnly*(self: Web3ProviderView, data: JsonNode): JsonNode =
+    let messageId = data["messageId"]
+    let messageType = "web3-send-async-callback"
+    let payloadId = data["payload"]["id"]
+    let rpcMethod = data["payload"]["method"].getStr()
```

Comment:
Then it could be read easily:
```suggestion
    let messageId = data.messageId
    let messageType = "web3-send-async-callback"
    let payloadId = data.payload.id
    let rpcMethod = data.payload.method
```
Also, we wouldn't need to set up these variables at all, bc they could be called directly later on without having to read the `JsonNode`.